src/toplevel.c

```diff
@@ -88,12 +88,15 @@ jl_value_t *jl_eval_module_expr(jl_expr_t *ex)
 {
     static arraylist_t module_stack;
     static int initialized=0;
+    jl_module_t *last_module;
+    jl_binding_t *b;
+    JL_GC_PUSH2(&last_module, &b);
```

Comment:
`jl_binding_t` is not a subtype of `jl_value_t` and should never be rooted in `JL_GC_PUSH` (it's rooted for the gc by the module that owns it)